### PR TITLE
Add new ol.geom.LineString#getCoordinateAt function

### LINKED ISSUE #3779: Exporting method for midpoint of a linestring
A method which returns a certain point of a linestring would be useful in many cases. There is already a function called "getFlatMidpoint" which returns the midpoint of the linestring.
I think this function should be exported and in addition it should provide a fraction as parameter in order to get any point on the linestring. For the midpoint i would call this method with "0.5" as argument.

### PR DESCRIPTION
**Pull request by fredj**
fixes #3779 

@openlayers/ol3 do you think it's a good addition to the API ?

### CODE REVIEW

**Comment by tschaub** [comment]
I think it is a useful addition (with minimal extra code).  We should talk about the name a bit.  Including "point" in the name might imply a `ol.geom.Point` is returned.  But `interpolatedCoordinate` is a mouthful.  We already have [`line.getCoordinateAtM()`](http://openlayers.org/en/master/apidoc/ol.geom.LineString.html#getCoordinateAtM).

Also wordy, but `line.getCoordinateAt(fraction)` *might* be a sensible name.

Or maybe we go for brevity with `line.interpolate(fraction)`.

Thoughts?

**Comment by fredj** [comment]
Function renamed to `getCoordinateAt`

File: src/ol/geom/linestring.js [hunk 1]
```diff
@@ -167,6 +167,21 @@ ol.geom.LineString.prototype.getCoordinates = function() {
 
 
 /**
+ * Return a point interpolated along the linestring, `fraction` is a number
```
Comment (tschaub) [inline src/ol/geom/linestring.js]: I think this should be broken into two sentences:

> Return the coordinate at the provided fraction along the linestring.  The `fraction` is
> a number between 0 and 1, where 0 is the start of the linestring and 1 is the end.

File: src/ol/geom/linestring.js [hunk 2]
```diff
@@ -167,6 +167,21 @@ ol.geom.LineString.prototype.getCoordinates = function() {
...
+ * between 0 and 1 representing fraction of total length of linestring.
+ * @param {number} fraction Fraction.
+ * @param {ol.Coordinate=} opt_dest Destination.
```
Comment (tschaub) [inline src/ol/geom/linestring.js]: I think "destination" is a bit confusing.  How about "Optional coordinate whose values will be modified.  If not provided, a new coordinate will be returned."?

**Comment by tschaub** [comment]
Thanks for updating this @fredj.  A couple minor doc comments.  This should make for pretty easy testing as well.

**Comment by fredj** [comment]
Unit tests added

**Comment by tschaub** [comment]
Looks great.  Please merge.

**Pull request merged**